src/Tests/Microsoft.NET.TestFramework/Assertions/CommandResultAssertions.cs

```diff
@@ -59,7 +59,14 @@ public AndConstraint<CommandResultAssertions> HaveStdOutContaining(string patter
                 .FailWith(AppendDiagnosticsTo($"The command output did not contain expected result: {pattern}{Environment.NewLine}"));
             return new AndConstraint<CommandResultAssertions>(this);
         }
-        
+
+        public AndConstraint<CommandResultAssertions> HaveStdOutContaining(Func<string, bool> predicate, string description = "")
+        {
+            Execute.Assertion.ForCondition(predicate(_commandResult.StdOut))
+                .FailWith(AppendDiagnosticsTo($"The command output did not contain expected result: {predicate.Method.Name} {description} {Environment.NewLine}"));
```

Comment:
I think we should not use `predicate.Method.Name` here.  It's not generally expected behavior to use the method name of a func you pass into a function, and if you use a lambda or anonymous delegate (perfectly reasonable to do), then this would be a compiler-generated method name.

Reply:
fixed, changed to just explicit pass the description.